src/main/java/net/neoforged/neoforge/common/NeoForgeMod.java

```diff
@@ -438,14 +464,14 @@ public NeoForgeMod(IEventBus modEventBus, Dist dist) {
 
         // Forge-provided datapack registries
         modEventBus.addListener((DataPackRegistryEvent.NewRegistry event) -> {
-            event.dataPackRegistry(ForgeRegistries.Keys.BIOME_MODIFIERS, BiomeModifier.DIRECT_CODEC);
-            event.dataPackRegistry(ForgeRegistries.Keys.STRUCTURE_MODIFIERS, StructureModifier.DIRECT_CODEC);
+            event.dataPackRegistry(NeoForgeRegistries.Keys.BIOME_MODIFIERS, BiomeModifier.DIRECT_CODEC);
+            event.dataPackRegistry(NeoForgeRegistries.Keys.STRUCTURE_MODIFIERS, StructureModifier.DIRECT_CODEC);
         });
         modEventBus.addListener(this::preInit);
         modEventBus.addListener(this::gatherData);
         modEventBus.addListener(this::loadComplete);
         modEventBus.addListener(this::registerFluids);
-        modEventBus.addListener(this::registerVanillaDisplayContexts);
+        modEventBus.addListener(EventPriority.HIGHEST, this::registerVanillaDisplayContexts);
```

Comment:
Why is this specifically at the highest priority?

Reply:
That's because we want to register the vanilla display contexts with the lowest possible IDs. (i.e. starting from 0)

Previously this couldn't be done like that because priorities on the `RegisterEvent` were ignored anyway.

Reply:
(Marking as resolved, feel free to object)

Reply:
We should probably add a comment to that effect there, so the reason for the lone priority use is easily accessible for posterity.